File: Data Structures & Algorithms/longest-common-prefix/submission-2.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {

        std::string res = "";

        for (int i = 0; i < strs[0].size(); ++i) {
            char c = strs[0][i];

            std::cout << c << std::endl;

            for (int j = 1; j < strs.size(); ++j) {
                if (strs[j][i] != strs[0][i]) {
                    return res;
                }
            }
            res += c;
        }
        return res;
    }
};

File: Data Structures & Algorithms/string-encode-and-decode/submission-2.py
class Solution:

    def encode(self, strs: List[str]) -> str:        

        fs = ""

        for s in strs:
            fs += str(len(s)) + "#" + s
        return fs

    def decode(self, s: str) -> List[str]:

        i = 0
        res = []

        while i < len(s):

            hp = s.find("#", i)

            size = int(s[i:hp])

            res.append(s[hp+1:hp+1+size])

            i = hp+1+size

        return res

